aten/src/ATen/functorch/BatchedTensorImpl.cpp

```diff
@@ -32,15 +32,36 @@ BatchedTensorImpl::BatchedTensorImpl(DispatchKeySet key_set, Tensor value, int64
 
 void BatchedTensorImpl::refreshTensorMetadata() {
   const auto public_dims = value_.dim() - 1;
-  const auto value_sizes = value_.sizes();
-  const auto value_strides = value_.strides();
-  sizes_and_strides_.resize(public_dims);
-  for (const auto dim : c10::irange(0, public_dims)) {
-    auto actual_dim = actualDim(dim, /*wrap_dim=*/false);
-    sizes_and_strides_.size_at_unchecked(dim) = value_sizes.at(actual_dim);
-    sizes_and_strides_.stride_at_unchecked(dim) = value_strides.at(actual_dim);
+  
+  // update size, strides and storage_offset
+  if (value_.unsafeGetTensorImpl()->has_symbolic_sizes_strides()){
+    // for tensor with symbolic size and strides
```

Comment:
For dynamic shapes support

Reply:
Why do we need new two different codepaths?  I would have expected a single codepath (the symbolic code path) to work

Reply:
Let's combine the two codepaths? (I don't remember if we came up with a reason not to, but my preference is to combine).

Reply:
This is done already.